repository.go

```diff
@@ -1030,25 +1030,47 @@ func (r repository) preload(cw contextWrapper, records slice, field string, quer
 		path                                             = strings.Split(field, ".")
 		targets, table, keyField, keyType, ddata, loaded = r.mapPreloadTargets(records, path)
 		ids                                              = r.targetIDs(targets)
-		query                                            = Build(table, append(queriers, In(keyField, ids...))...)
+
+		inClauseLength = 999 // TODO: can this value come from the adapter, as it is dbms specific?
 	)
 
-	if len(targets) == 0 || loaded && !bool(query.ReloadQuery) {
-		return nil
-	}
+	// Create separate queries if the amount of ids is more than inClauseLength.
+	for {
+		if len(ids) == 0 {
+			break
+		}
 
-	var (
-		cur, err = cw.adapter.Query(cw.ctx, r.withDefaultScope(ddata, query, false))
-	)
+		// necessary check to avoid slicing beyond
+		// slice capacity
+		if len(ids) < inClauseLength {
+			inClauseLength = len(ids)
+		}
 
-	if err != nil {
-		return err
-	}
+		idsChunk := ids[0:inClauseLength]
+		ids = ids[inClauseLength:]
+
+		query := Build(table, append(queriers, In(keyField, idsChunk...))...)
+		if len(targets) == 0 || loaded && !bool(query.ReloadQuery) {
+			return nil
+		}
+
+		var (
+			cur, err = cw.adapter.Query(cw.ctx, r.withDefaultScope(ddata, query, false))
+		)
 
-	scanFinish := r.instrumenter.Observe(cw.ctx, "rel-scan-multi", "scanning all records to multiple targets")
-	defer scanFinish(nil)
+		if err != nil {
+			return err
+		}
 
-	return scanMulti(cur, keyField, keyType, targets)
+		scanFinish := r.instrumenter.Observe(cw.ctx, "rel-scan-multi", "scanning all records to multiple targets")
+		err = scanMulti(cur, keyField, keyType, targets)
```

Comment:
Maybe this can be added as assertion in the unit test?

Reply:
Ok, as long as the keys do not re-occur in the subsequent calls it works just fine, as each scanMulti call only modifies the occurred keys.  
This works as long as  each call uses unique keys which is here the case as r.targetIDs returns unique keys.

So it should be fine here. Tell me if you think otherwise.

I added a test to scanMulti for this.